packages/components/switch/src/Switch.test.tsx

```diff
@@ -101,4 +103,49 @@ describe('Switch', () => {
       expect(screen.getByRole('switch')).toBeChecked()
     })
   })
+
+  describe('with FormField', () => {
+    it('should render with label and name', () => {
+      render(
+        <FormField name="agreement">
+          <FormField.Label asChild>
+            <p>Agreement</p>
+          </FormField.Label>
+
+          <Switch />
+        </FormField>
+      )
+
+      expect(screen.getByRole('switch', { name: 'Agreement' })).toBeInTheDocument()
+    })
+
+    it('should render aria-attributes following FormField implementation', () => {
+      render(
+        <FormField name="agreement" isRequired isInvalid>
+          <FormField.Label asChild>
+            <p>Agreement</p>
+          </FormField.Label>
+
+          <Switch />
+
+          <FormField.ErrorMessage>Agreement is required</FormField.ErrorMessage>
```

Comment:
Maybe would be nice to also test the helper message ?

Reply:
I'm not sure of it. The FormField components (ErrorMessage, ...) should already be tested in related package. It would be duplicate to do it again here, don't you think?